docs/crystal_design.md

```diff
@@ -0,0 +1,187 @@
+# Metadata Management (Crystal)
+
+**Author(s)**: [Pravin Ranjan](https://github.com/PravinRanjan10)
+
+## Motivation and background
+
+Metadata gives basic information about data—including type, time of creation, size of object/buckets, storage class, and more for cloud object storages. Metadata management is the business discipline of managing the metadata about data. It can play key roles during decision making activity. Thus metadata management is required nowadays.
+
+A user may provision s3-cloud buckets using some solution like strato(multi-cloud project from SODAFOUNDATION) and/or manually. To get and manage metadata of all buckets/objects provisioned by manually and automatically, another solution required. Because, there would be no metadata available for manually provisioned buckets.
+
+### Goals
+Develop a Metadata Management Service which can discover, harvest, and manage all Externally orchestrated S3 buckets and its corresponding metadata from one place. It should perform following operations:
+Access the Metadata
+Acquire the Metadata
+Store the Metadata
+And, Process the Metadata(search, classify, optimize etc.)
+
+### Non-Goals
+The ability to manage nfs, smb etc will be handled in future
+Unification of metadata interfaces will be handled in future
+Data mover functionality is out of scope of this analysis and design
+API gateway integration will be handled in future
+
+
+### Functional Requirements
+* **Discovery and Extraction**: Automate metadata harvesting of externally orchestrated S3 buckets from multiple Cloud vendors. This can be done as:
+  * Manual Sync
```

Comment:
What is sync? For each bucket, are we going to replace the old metadata with new metadata? Are we going to keep copies of changed metadata (maybe in a database log) so that older version can be queried to find changes in metadata? If we support metadata change detection, that may be a feature nobody has, though I don't know if this is a user requirement. Change detection could be useful during audits